configuration/config2/XSDSchemaParser.cpp

```diff
@@ -722,50 +715,38 @@ std::shared_ptr<SchemaValue> XSDSchemaParser::getSchemaValue(const pt::ptree &at
         std::shared_ptr<SchemaType> pType = getType(attr.get_child("xs:simpleType", pt::ptree()), false);
         if (!pType->isValid())
         {
-            throw(ParseException("Attribute " + attrName + " does not have a valid type"));
+            throw(ParseException("Attribute " + m_pSchemaItem->getProperty("name") + "[@" + attrName + "] does not have a valid type"));
```

Comment:
minor, we could also output the invalid type

Reply:
The code in this case is when no type name was given, so the parser is looking for an adhoc type definition in an <xs:simpleType> child section defined in the attribute definition. The error message is generated when none is found. So, in essence we don't have a name for the invalid type, the attribute truly has no valid type.